internal/provider/bootstrap.go

```diff
@@ -34,23 +32,23 @@ func Bootstrap(e *pluggable.Event) pluggable.EventResponse {
 	}
 
 	c := &config.Config{}
-	providerConfig := &providerConfig.Config{}
+	prvConfig := &providerConfig.Config{}
 	err = config.FromString(cfg.Config, c)
 	if err != nil {
 		return ErrorEvent("Failed reading JSON input: %s input '%s'", err.Error(), cfg.Config)
 	}
 
-	err = config.FromString(cfg.Config, providerConfig)
+	err = config.FromString(cfg.Config, prvConfig)
 	if err != nil {
 		return ErrorEvent("Failed reading JSON input: %s input '%s'", err.Error(), cfg.Config)
 	}
 	// TODO: this belong to a systemd service that is started instead
 
-	p2pBlockDefined := providerConfig.P2P != nil
-	tokenNotDefined := ((p2pBlockDefined && providerConfig.P2P.NetworkToken == "") || !p2pBlockDefined)
-	skipAuto := (p2pBlockDefined && !providerConfig.P2P.Auto.IsEnabled())
+	p2pBlockDefined := prvConfig.P2P != nil
```

Comment:
This will always be true because the providerConfig.Config.P2P is never gonna be nil as its a pointer!

Reply:
wait, pointers can be nil

Reply:
But when the provider config gets initialized, it creates a new P2P block no? I encountered this before and I thought it worked like that...

Reply:
Maybe the other way around?  https://go.dev/play/p/qdaBKnHBK1A